base/compiler/bootstrap.jl

```diff
@@ -5,10 +5,20 @@
 # especially try to make sure any recursive and leaf functions have concrete signatures,
 # since we won't be able to specialize & infer them at runtime
 
-let fs = Any[typeinf_ext, typeinf, typeinf_edge, pure_eval_call, run_passes],
```

Comment:
I noticed that besides reordering, `pure_eval_call` has been removed. How come that it isn't needed anymore?

Reply:
`pure_eval_call` can be easily inferred from the entries like `typeinf`, because we make sure `Core.Compiler` is very type stable.